src/ccall.cpp

```diff
@@ -233,10 +233,11 @@ static GlobalVariable *emit_plt_thunk(
     else {
         // musttail support is very bad on ARM, PPC, PPC64 (as of LLVM 3.9)
         // Known failures includes vararg (not needed here) and sret.
```

Comment:
possible improvement?
```suggestion
        // Known failures includes vararg (not needed here) and sret.
        ret->setTailCall();
```

Is this only M1, or all AArch64?

Reply:
This is m1 only.  I wonder if it's this https://github.com/llvm/llvm-project/blob/d1cd64ffdd832220dbe1829c2f09b880be67be31/llvm/test/CodeGen/AArch64/GlobalISel/call-translator-musttail.ll, this is the only aarch64-apple exclusive thing related to musttail.

That might be a more robust solution for sure. Does doing the ccall as a tail call have some performance benefit?